File: session-03-who-owns-the-main-loop/python/order/service.py
from typing import Optional

from order.model import Order
from order.repository import OrderRepository


class OrderService:
    def __init__(self, repo: OrderRepository):
        self.repo = repo

    def list_orders(self) -> list[Order]:
        return self.repo.find_all()

    def find_order(self, order_id: int) -> Optional[Order]:
        return self.repo.find_by_id(order_id)

File: session-03-who-owns-the-main-loop/go/order/service.go
package order

type OrderService struct {
	repo OrderRepository
}

func NewOrderService(repo OrderRepository) *OrderService {
	return &OrderService{repo: repo}
}

func (s *OrderService) ListOrders() []Order {
	return s.repo.FindAll()
}

func (s *OrderService) FindOrder(id int) *Order {
	return s.repo.FindByID(id)
}
